azext_iot/_params.py

```diff
@@ -63,7 +63,8 @@
     type=str,
     nargs="?",
     choices=["0", "1"],
-    help="Quality of Service. 0 = At most once, 1 = At least once. 2 (Exactly once) is not supported.",
+    help="Quality of Service. 0 = At most once, 1 = At least once. 2 (Exactly once) is not supported."
+    "This command parameter has been deprecated and will be removed in a future release.",
```

Comment:
For deprecating we use `deprecate_info` like so

![image](https://user-images.githubusercontent.com/28658112/118163033-55df4180-b3d6-11eb-9f04-7be1aac55224.png)

And the deprecation message needs to be in a released version prior to the version where the functionality stops working or is removed.

Reply:
Point being we will need to add this in a different branch

Reply:
As discussed, I am removing QoS in this PR. The deprecation will be done in a separate version which will be released before these changes are released